recipes/atk/all/conanfile.py

```diff
@@ -1,102 +1,130 @@
-from conans import ConanFile, Meson, tools
-from conans.errors import ConanInvalidConfiguration
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.apple import fix_apple_shared_install_name
+from conan.tools.build import cross_building
+from conan.tools.env import VirtualBuildEnv, VirtualRunEnv
+from conan.tools.files import apply_conandata_patches, copy, export_conandata_patches, get, replace_in_file, rm, rmdir
+from conan.tools.gnu import PkgConfigDeps
+from conan.tools.layout import basic_layout
+from conan.tools.meson import Meson, MesonToolchain
+from conan.tools.microsoft import is_msvc_static_runtime
 import os
-import glob
 
-required_conan_version = ">=1.29"
+required_conan_version = ">=1.53.0"
+
 
 class AtkConan(ConanFile):
     name = "atk"
     description = "set of accessibility interfaces that are implemented by other toolkits and applications"
-    topics = ("conan", "atk", "accessibility")
+    topics = ("accessibility",)
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "https://www.atk.org"
     license = "LGPL-2.1-or-later"
-    generators = "pkg_config"
 
     settings = "os", "arch", "compiler", "build_type"
     options = {
         "shared": [True, False],
         "fPIC": [True, False],
-        }
+    }
     default_options = {
         "shared": False,
         "fPIC": True,
-        }
-
-    _source_subfolder = "source_subfolder"
-    _build_subfolder = "build_subfolder"
+    }
 
-    exports_sources = "patches/**"
+    def export_sources(self):
+        export_conandata_patches(self)
 
     def config_options(self):
-        if self.settings.os == 'Windows':
+        if self.settings.os == "Windows":
             del self.options.fPIC
 
-    def build_requirements(self):
-        self.build_requires('meson/0.60.2')
-        self.build_requires('pkgconf/1.7.4')
-
-    def requirements(self):
-        self.requires('glib/2.73.0')
-
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
-        del self.settings.compiler.libcxx
-        del self.settings.compiler.cppstd
+            self.options.rm_safe("fPIC")
+        self.settings.rm_safe("compiler.cppstd")
+        self.settings.rm_safe("compiler.libcxx")
         if self.options.shared:
             self.options["glib"].shared = True
 
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("glib/2.75.2")
+
+    def package_id(self):
+        self.info.requires["glib"].full_package_mode()
+
```

Comment:
I don't disagree but it's out of scope of this PR.  It was already there. I've moved this method to sort by order of execution.

Reply:
@uilianries maybe you want to open a PR to remove this `package_id()` logic? It may be a good idea adding some words about it in documentation. I know some people out there would like to track boost & glib minor version in package id of downstream.